syntax = "proto3";

package surround360.proto;

message ProjectSphericalArgs {
  int32 eqr_height = 1;
  int32 eqr_width = 2;
  string camera_rig_path = 3;
}

message TemporalOpticalFlowArgs {
  string flow_algo = 1;
  string camera_rig_path = 2;
}

message RenderStereoPanoramaChunkArgs {
  int32 eqr_height = 1;
  int32 eqr_width = 2;
  string camera_rig_path = 3;
  string flow_algo = 4;
  float zero_parallax_dist = 5;
  float interpupilary_dist = 6;
}

message ConcatPanoramaChunksArgs {
  int32 eqr_height = 1;
  int32 eqr_width = 2;
  int32 final_eqr_height = 7;
  int32 final_eqr_width = 8;
  string camera_rig_path = 3;
  float zero_parallax_dist = 4;
  float interpupilary_dist = 5;
  bool left = 6;
}
